crates/spk-solve/crates/graph/src/graph.rs

```diff
@@ -1374,23 +1374,26 @@ impl State {
 
     /// Get a mapping of pkg name -> merged request for the unresolved
     /// PkgRequests in this state
-    pub fn get_unresolved_requests(&self) -> &HashMap<PkgNameBuf, PkgRequest> {
-        self.cached_unresolved_pkg_requests.get_or_init(|| {
-            let mut unresolved: HashMap<PkgNameBuf, PkgRequest> = HashMap::new();
-
-            for req in self.pkg_requests.iter() {
-                if unresolved.contains_key(&req.pkg.name) {
-                    continue;
-                }
-                if self.get_current_resolve(&req.pkg.name).is_err() {
-                    unresolved.insert(
-                        req.pkg.name.clone(),
-                        self.get_merged_request(&req.pkg.name).unwrap(),
-                    );
+    pub fn get_unresolved_requests(&self) -> Result<&HashMap<PkgNameBuf, PkgRequest>> {
+        match self.cached_unresolved_pkg_requests.get() {
+            Some(hm) => Ok(hm),
+            None => {
```

Comment:
could you not use `get_or_try_init` instead of this additional matching?

Reply:
It's marked as nightly experimental. I tend to avoid those. But I've updated this to use it.

Reply:
in the `std::cell` implementation, yes but we are using the `once_cell` crate in which it's stable - otherwise, I would agree to avoid it